services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/UseGlobalEndpointResolver.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.internal.endpoints;
+
+import java.util.function.Supplier;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.awscore.client.config.AwsClientOption;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientOption;
+import software.amazon.awssdk.profiles.ProfileFile;
+import software.amazon.awssdk.profiles.ProfileProperty;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.regions.ServiceMetadataAdvancedOption;
+import software.amazon.awssdk.utils.Lazy;
+
+/**
+ * Note: Most of the logic is copied from EnhancedS3ServiceMetadata
+ * // TODO: should probably refactor that class to reuse the same logic
+ */
+@SdkInternalApi
+public class UseGlobalEndpointResolver {
+    private static final String REGIONAL_SETTING = "regional";
+    private final Lazy<Boolean> useUsEast1RegionalEndpoint;
+    private final SdkClientConfiguration sdkClientConfiguration;
+
+    public UseGlobalEndpointResolver(SdkClientConfiguration config) {
+        sdkClientConfiguration = config;
+        String defaultS3UsEast1RegionalEndpointFromSmartDefaults =
+            config.option(ServiceMetadataAdvancedOption.DEFAULT_S3_US_EAST_1_REGIONAL_ENDPOINT);
+        this.useUsEast1RegionalEndpoint =
+            new Lazy<>(() -> useUsEast1RegionalEndpoint(() ->config.option(SdkClientOption.PROFILE_FILE), () -> config.option(SdkClientOption.PROFILE_NAME),
+                                                        defaultS3UsEast1RegionalEndpointFromSmartDefaults));
```

Comment:
Nit: Can this statement be formatted so it's easier to read? Short the amount of text or line up the parameters?

The original has ` this.useUsEast1RegionalEndpoint = new Lazy<>(() -> useUsEast1RegionalEndpoint(profileFile, profileName, config))` which was easier to parse

Reply:
done